src/sentry/integrations/slack/requests/base.py

```diff
@@ -115,20 +104,20 @@ def integration(self) -> RpcIntegration:
         return self._integration
 
     @property
-    def channel_id(self) -> str:
-        return get_field_id(self.data, "channel")
+    def channel_id(self) -> str | None:
```

Comment:
i'm not sure if it's okay if this is `None`

for linking (and similarly for unlinking) a user we have this check https://github.com/getsentry/sentry/blob/121726dc100a21249e177200f1054f8f5a30fbbb/src/sentry/integrations/slack/webhooks/base.py#L64-L65

we assume it's not `None` for linking (and unlinking) a team
https://github.com/getsentry/sentry/blob/121726dc100a21249e177200f1054f8f5a30fbbb/src/sentry/integrations/slack/views/link_team.py#L39-L53
https://github.com/getsentry/sentry/blob/121726dc100a21249e177200f1054f8f5a30fbbb/src/sentry/integrations/slack/webhooks/event.py#L46-L57

Reply:
You're basically undoing [this change](https://github.com/getsentry/sentry/pull/29912). It seems like from before that PR channel was never nullable while team and user were optional. So I think you're right.